packages/compass-components/src/hooks/use-copy-paste-context-menu.tsx

```diff
@@ -0,0 +1,218 @@
+import React, { useCallback, useState, useEffect } from 'react';
+
+import { useContextMenuGroups } from '../components/context-menu';
+
+const NON_TEXT_INPUT_TYPES = [
+  'checkbox',
+  'radio',
+  'submit',
+  'button',
+  'image',
+  'file',
+  'color',
+  'range',
+];
+const SELECTION_EVENTS = [
+  'selectionchange',
+  'focusin',
+  'focusout',
+  'input',
+  'keyup',
+  'mouseup',
+  'contextmenu',
+];
+
+const isEditableTextElement = (
+  element: Element | null
+): element is HTMLElement => {
+  if (!element) return false;
+
+  const tagName = element.tagName.toLowerCase();
```

Comment:
I'm curious if there's a reason you use `toLowerCase` here?

Reply:
Nope, tagName capitalization is still a mystery to me after reading the docs. https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName
I like your suggestion of using instanceof here. It's a lot nicer. Removed this